PyInstaller/utils/hooks/__init__.py

```diff
@@ -984,3 +986,61 @@ def relpath_to_config_or_make(filename):
     # Relative path in the dist directory.
     prefix = _find_prefix(filename)
     return os.path.relpath(os.path.dirname(filename), prefix)
+
+
+def get_typelibs(module, version):
+    datas = []
+    statement = """
+import gi
+gi.require_version("GIRepository", "2.0")
+from gi.repository import GIRepository
+print(GIRepository.Repository.get_search_path())"""
+    typelibs_path = eval_statement(statement)[0]
+    pattern = os.path.join(typelibs_path, module + '*' + version + '*')
+    for f in glob.glob(pattern):
+        datas.append(gir_library_path_fix(f))
+    return datas
+
+
+def gir_library_path_fix(path):
+    import subprocess
+
+    path = os.path.abspath(path)
+    common_path = os.path.commonprefix([sys.prefix, path])
+    gir_path = os.path.join(common_path, 'share', 'gir-1.0')
+
+    typelib_name = os.path.basename(path)
+    gir_name = os.path.splitext(typelib_name)[0] + '.gir'
+
+    gir_file = os.path.join(gir_path, gir_name)
+
+    if is_darwin:
+        if os.path.exists(gir_path):
+            if os.path.exists(gir_file):
```

Comment:
@htgoebel I'm not sure I understand what you would like me to change about the `if` statements.

I'm not sure exactly what I want to do about the return. On one hand having the function return data in some instances and not others seems like a bad idea but on the other hand this is exactly want I intended or at least it produces what I intended which is to make the build fail if the gir file cannot be found.

Thoughts appreciated.